force-app/main/default/lwc/rd2RecurringDonation/rd2RecurringDonation.html

```diff
@@ -0,0 +1,171 @@
+                        </td>
+                        <td key={item.status} data-label={labels.lblStatus}>
+                            <div class={rowClasses} style={fixedWidth} title={labels.lblStatus}>
+                                {item.status}</div>
+                        </td>
+                        <td key={item.firstDonation} data-id={item.id} class={tdClasses} data-label={labels.firstDonation}>
+                            <div class={rowClasses} style={fixedWidth} title={labels.firstDonation}>
+                                {item.firstDonation}
+                            </div>
+                        </td>
+                        <td key={item.recurringDonation.npe03__Last_Payment_Date__c} data-id={item.id} class={tdClasses} data-label={labels.mostRecentDonation}>
+                            <div class={rowClasses} style={fixedWidth} title={labels.mostRecentDonation}>
+                                {item.recurringDonation.npe03__Last_Payment_Date__c}</div>
+                        </td>
+                        <td key={item.nextDonation} data-id={item.id} class={tdClasses}  data-label={labels.nextDonation}>
+                            <div class={rowClasses} style={fixedWidth}> 
+                                {item.nexDonationFormatFirstElement} </br>
+                                {item.nexDonationFormatSecondElement}
+                            </div>
+                        </td>
+                        <td key={item.lastModified} data-id={item.id} class={tdClasses}  data-label={labels.lastModified}>
+                            <div class={rowClasses} style={fixedWidth} title="Last Modified">
+                                {item.lastModified}</div>
+                        </td>
+                        <td data-label="Actions">
+                            <lightning-button-menu class="dv-dynamic-mobile" menu-alignment="auto" alternative-text="Action menu">
+                                <template for:each={actions} for:item="action">
+                                    <p key={action.name} data-recid={action.name}>
+                                        <lightning-menu-item label={action.label} onclick={handleRowAction} data-action={action.name} data-recordid={item.recurringDonation.Id}>
+                                        </lightning-menu-item>
+                                    </p>
+                                </template>
+                            </lightning-button-menu>
+                        </td>
+                        <td class={viewMore} data-label={labels.RD2_ViewMoreDetails} data-viewid={item.id} onclick={toggleView}>
+                        </td>
+                    </tr>
+                </template>
+            </tbody>
+        </table>
+    </div>
+    <c-update-payment-method-modal current-record={currentRecord} open-update-payment-method={openUpdatePaymentMethod} onclose={handleClose}></c-update-payment-method-modal>
```

Comment:
we have separated the logic since the requirement of each modal is different, internally in each of those components the currentRecord is being used to complete the required functionality, you can take a look at one of those implementations in these PRs
https://github.com/SalesforceFoundation/NPSP/pull/6982
https://github.com/SalesforceFoundation/NPSP/pull/6981/files